src/main.c

```diff
@@ -214,6 +215,11 @@ int main(int argc, char **argv)
 		if(bind(fd, (struct sockaddr*)&addr, sizeof addr) < 0) die();
 	}
 
+	// create only the last directory not recursive mode
+    	struct stat st = {0};
+    	if (stat(dirname(socket_path), &st) == -1) {
+           mkdir(dirname(socket_path), 0755);
+    	}
```

Comment:
This isn't in the right place, directory creation should happen before the `bind()` attempt. We should also consider only doing it in case `bind` returns an error indicating the directory doesn't exist, instead of using `stat` to check for directory existence (`access` or `readlink` would be lighterweight options, btw).